packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -17,200 +17,230 @@ import {
+			);
 		}
 
 		beforeEach(async function () {
 			provider = getTestObjectProvider({ syncSummarizer: true });
+			if (provider.driver.type !== "local") {
+				this.skip();
+			}
+			settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+
+			const configProvider = mockConfigProvider(settings);
+			testContainerConfig = {
+				runtimeOptions: {
+					summaryOptions: {
+						summaryConfigOverrides: {
+							state: "disabled",
+						},
+					},
+					gcOptions,
+				},
+				loaderProps: { configProvider },
+			};
+
+			summarizerContainerConfig = {
+				runtimeOptions: { gcOptions },
+				loaderProps: { configProvider },
+			};
+		});
+
+		afterEach(() => {
+			settings = {};
 		});
 
-		it(`A summary has a datastore and blob referenced, but trailing ops unreferenced them ${
-			tombstoneEnabled ? "after sweep timeout" : "before sweep timeout"
-		}`, async () => {
+		it(`Trailing ops ${
+			beforeSweepTimeout ? "before sweep timeout" : "after sweep timeout"
+		} makes data store ${
+			transitionToReference ? "referenced" : "unreferenced"
+		} without deleting it`, async () => {
 			const mainContainer = await provider.makeTestContainer(testContainerConfig);
-			const mainDefaultDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
+			const mainDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
 			await waitForContainerConnection(mainContainer);
 
-			// Create a data store and blob.
+			const newDataStoreKey = "datastore";
+			// Create a data store and reference it.
 			const newDataStore =
-				await mainDefaultDataStore._context.containerRuntime.createDataStore(
-					TestDataObjectType,
-				);
+				await mainDataStore._context.containerRuntime.createDataStore(TestDataObjectType);
 			assert(newDataStore.entryPoint !== undefined, `Should have a handle`);
-			const blobContents = "Blob contents";
-			const blobHandle = await mainDefaultDataStore._runtime.uploadBlob(
-				stringToBuffer(blobContents, "utf-8"),
+			const newTestDataObject = (await newDataStore.entryPoint.get()) as ITestDataObject;
+			mainDataStore._root.set(newDataStoreKey, newDataStore.entryPoint);
+
+			// Update the reference state. The data store should start in the opposite state of "transitionToReference"
+			// and the trailing op will transition it to that state.
+			updateDataStoreReferenceState(
+				mainDataStore,
+				newTestDataObject,
+				newDataStoreKey,
+				!transitionToReference,
 			);
 
 			// Create a summarizer
-			const { summarizer: mainSummarizer } = await createSummarizer(provider, mainContainer, {
-				runtimeOptions: { gcOptions },
-				loaderProps: { configProvider },
-			});
-
-			// Reference datastore and blob
-			mainDefaultDataStore._root.set("datastore", newDataStore.entryPoint);
-			mainDefaultDataStore._root.set("blob", blobHandle);
+			const { summarizer: mainSummarizer } = await createSummarizer(
+				provider,
+				mainContainer,
+				summarizerContainerConfig,
+			);
 
```

Comment:
Sharing a personal practice I've been doing - for asserts like this that aren't the main point of the test, prefix them like this: `"PRECONDITION: Should have a handle"`